src/foam/nanos/ruler/RuleActionMergeDecorator.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @license
+ * Copyright 2021 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.nanos.ruler',
+  name: 'RuleActionMergeDecorator',
+
+  documentation: `
+    Merge calls occurring within mergeDelay period to the delegate rule action.
+
+    Usage: Add RuleActionMergeDecorator to the 'action' of an async rule to
+    merge repeated calls to the underlying rule action. For example,
+
+      {
+        class: 'foam.nanos.ruler.Rule',
+        ...,
+        async: true,
+        action: {
+          class: 'foam.nanos.ruler.RuleActionMergeDecorator',
+          mergeDelay: 1000,
+          delegate: {
+            class: 'MyRuleAction'
+          }
+        }
+      }
+  `,
+
+  implements: [ 'foam.nanos.ruler.RuleAction' ],
+
+  javaImports: [
+    'foam.nanos.logger.Logger',
+    'java.util.Timer',
+    'java.util.TimerTask',
+    'java.util.concurrent.ConcurrentHashMap'
+  ],
+
+  properties: [
+    {
+      class: 'FObjectProperty',
+      of: 'foam.nanos.ruler.RuleAction',
+      name: 'delegate'
+    },
+    {
+      class: 'Int',
+      name:  'mergeDelay',
+      units: 'ms'
+    },
+    {
+      class: 'Object',
+      javaType: 'java.util.Timer',
+      name: 'timer',
+      javaFactory: 'return new Timer();'
+    },
+    {
+      class: 'Map',
+      name: 'taskQueue',
+      javaFactory: 'return new ConcurrentHashMap();'
+    }
+  ],
+
+  methods: [
+    {
+      name: 'applyAction',
+      javaCode: `
+        if ( ! rule.getAsync()
```

Comment:
I know that this is a very minor complaint, but, why not just put the whole expression on one line? It still falls well under 80 characters and is much easier to read.